packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -0,0 +1,225 @@
+import { LightRPC, LightScriptFilter } from '../../utils/ckb-rpc'
+import HexUtils from 'utils/hex'
+
+interface SyncQueueParam {
+  script: CKBComponents.Script
+  scriptType: CKBRPC.ScriptType
+  blockRange: [HexString, HexString]
+  cursor?: HexString
+}
+
+export default class LightConnector extends Connector<CKBComponents.Hash> {
+  private lightRpc: LightRPC
+  private indexer: CkbIndexer
+  private addressMetas: AddressMeta[]
+  private syncQueue: QueueObject<SyncQueueParam> = queue(this.syncNextWithScript.bind(this), 1)
+  private indexerQueryQueue: QueueObject<LumosCellQuery> | undefined
+  private pollingIndexer: boolean = false
+  private syncInQueue: Map<
+    CKBComponents.Hash,
+    { blockStartNumber: number; blockEndNumber: number; cursor?: string }
+  > = new Map()
+
+  public readonly blockTipsSubject: Subject<BlockTips> = new Subject<BlockTips>()
+  public readonly transactionsSubject = new Subject<{ txHashes: CKBComponents.Hash[]; params: CKBComponents.Hash }>()
+
+  constructor(addresses: Address[], nodeUrl: string) {
+    super()
+    this.indexer = new CkbIndexer(nodeUrl, nodeUrl)
+    this.lightRpc = new LightRPC(nodeUrl)
+    this.addressMetas = addresses.map(address => AddressMeta.fromObject(address))
+
+    this.indexerQueryQueue = queue(this.collectLiveCellsByScript)
+  }
+
+  private async synchronize() {
+    if (this.syncQueue.idle()) {
+      await this.subscribeSync()
+      const syncScripts = await this.lightRpc.getScripts()
+      const syncStatusMap = await SyncProgressService.getAllSyncStatusToMap()
+      syncStatusMap.forEach(v => {
+        if (v.cursor && !this.syncInQueue.has(v.hash)) {
+          this.syncQueue.push({
+            script: {
+              codeHash: v.codeHash,
+              hashType: v.hashType,
+              args: v.args
+            },
+            blockRange: [HexUtils.toHex(v.blockStartNumber), HexUtils.toHex(v.blockEndNumber)],
+            scriptType: v.scriptType,
+            cursor: v.cursor
+          })
+        }
+      })
+      syncScripts.forEach(syncScript => {
+        const scriptHash = scriptToHash(syncScript.script)
+        const syncStatus = syncStatusMap.get(scriptHash)
+        if (
+          syncStatus &&
+          !this.syncInQueue.has(scriptHash) &&
+          !syncStatus.cursor &&
+          syncStatus.blockEndNumber < parseInt(syncScript.blockNumber)
+        ) {
+          this.syncQueue.push({
+            script: syncScript.script,
+            blockRange: [HexUtils.toHex(syncStatus.blockEndNumber), syncScript.blockNumber],
+            scriptType: syncScript.scriptType,
+            cursor: undefined
+          })
+        }
+      })
+    }
+  }
+
+  private async subscribeSync() {
+    const minSyncBlockNumber = await SyncProgressService.getMinBlockNumber()
+    const header = await this.lightRpc.getTipHeader()
+    this.blockTipsSubject.next({
+      cacheTipNumber: minSyncBlockNumber,
+      indexerTipNumber: parseInt(header.number, 16)
```

Comment:
Why not use `+header.number` to transform an integer from hex format to number format

`parseInt` accepts unrecognizable char, as `parseInt('0xfzz', 16)` will be `15`